gamemodes/cinema/gamemode/modules/scoreboard/cl_playerlist.lua

```diff
@@ -153,48 +216,116 @@ end
-    self.Ping:SetPos(402, 11)
-    self.Country = vgui.Create("DImage", self)
-    self.Country:SetSize(16, 11)
-    self.Country:SetPos(388, 18)
-    self.Country:SetImage("countries/us.png")
-    self.Country:CenterVertical()
+    self.Location = Label("Unknown", self)
+    self.Location:SetFont("ScoreboardLocation")
+    self.Location:SetColor(Color(255, 255, 255, 80))
+    self.Location:SetPos(0, 8)
+
+    self.FriendIcon = vgui("DImage", self, function(p)
+        p:SetSize(16, 16)
+        p:SetPos(30, 27)
+        p:SetZPos(2)
+        p:SetImage("chev/icon/friend.png")
+    end)
+
+    --holder for all these little buttons so we can dock them away from the edge
+    vgui("DPanel", self, function(p)
+        p.Paint = noop
+        p:SetWide(20)
+        p:Dock(RIGHT)
+        p:DockMargin(2, 2, 2, 2)
+
+        self.Mute = vgui("DImageButton", function(p)
+            p:SetSize(20, 20)
+            p:Dock(BOTTOM)
+            p:SetImage("icon32/unmuted.png")
+            p:SetToolTip("Toggle Voice Mute")
+        end)
+
+        self.ChatMute = vgui("DImageButton", function(p)
+            p:SetSize(20, 20)
+            p:Dock(TOP)
+            p:SetImage("theater/chatunmuted.png")
+            p:SetToolTip("Toggle Chat Mute")
+        end)
+    end)
+
+    vgui("DPanel", self, function(p)
+        p.Paint = noop
+        p:Dock(RIGHT)
+        p:DockMargin(2, 2, 2, 2)
+        p:DockPadding(2, 2, 2, 2)
+        p:InvalidateParent(true)
+
+        self.Ping = vgui("ScoreboardPlayerPing", function(p)
+            p:Dock(TOP)
+        end)
+
+        self.Country = vgui("DImageButton", function(p)
+            p:SetSize(16, 11)
+            p:SetText("")
+            p:CenterHorizontal()
+            p:AlignBottom(4)
+            p:AlignRight(4)
+            p:SetImage("countries/us.png")
+        end)
+
+        p:InvalidateLayout()
+        p:SizeToChildren(true, true)
+    end)
+
+    vgui("DPanel", self, function(p)
+        p.Paint = noop
+        p:SetTall(20)
+        p:Dock(RIGHT)
+        local cgap = (self:GetTall() - p:GetTall()) / 2
+        p:DockMargin(0, cgap, SS_COMMONMARGIN, cgap)
+
+        if (showAFKs or LocalPlayer():IsStaff()) then
+            self.AFK = vgui("DImage", function(p)
+                p:SetSize(16, 16)
+                p:DockMargin(2, 2, 2, 2)
+                p:Dock(RIGHT)
+                p:SetImage("icon16/time.png")
+                p:CenterVertical()
+            end)
+        end
+
+        p:InvalidateLayout()
+        p:SizeToChildren(true, true)
+        p:SetWide(256)
+    end)
+end
+
+local function GetCountryData()
+    http.Fetch("https://raw.githubusercontent.com/lukes/ISO-3166-Countries-with-Regional-Codes/master/all/all.json", function(str)
+        local tab = util.JSONToTable(str)
+        local newtab = {}
+
+        for k, v in pairs(tab) do
+            newtab[string.lower(v["alpha-2"])] = v
+        end
+
+        CountryData = newtab
+    end)
 end
 
+GetCountryData()
```

Comment:
Can you not do this extraneous http request until someone has actually moused over the thing or whatever

Reply:
i don't think there's an event for when tooltips appear